tests/unit-core/test-promise-callback.c

```diff
@@ -0,0 +1,283 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+
+#include "test-common.h"
+
+typedef enum
+{
+  C = JERRY_PROMISE_CALLBACK_CREATE, /**< same as JERRY_PROMISE_CALLBACK_CREATE with undefined value */
```

Comment:
Could we have a bit more descriptive names? like:
``` 
C -> CREATE,
RS -> RESOLVE,
RJ -> REJECT,
BR -> BEFORE_REACTION,
BRS-> BEFORE_RESOLVE (?)
```

At the moment I don't see the logic in the name BRS, BRJ, ARS, ARJ at the moment.
Also I was wondering if we need these at all. Can't we just use the API names and only define the two extra values (CP/E) ?

Reply:
There are lot of static arrays bellow. If we would use long names, they would go over the 120 char limit. Then we need to split them into multiple lines, where only one value would go in one line. This would look ugly.